firmware/src/boards/cynthion_d11/fpga.c

```diff
@@ -11,25 +11,41 @@
 #include "apollo_board.h"
 #include "jtag.h"
 #include "fpga.h"
+#include "board_rev.h"
 
 
+#ifdef BOARD_HAS_FPGA_SYSCFG_PINS
+static bool board_has_fpga_initn(void)
+{
+#if (_BOARD_REVISION_DETECT_ == 1)
+	uint16_t revision = get_board_revision();
+    const uint8_t major = revision >> 8;
+    const uint8_t minor = revision & 0xFF;
+    return major == 1 && minor > 2;
+#endif
+    return true;
+}
+#endif
+
 /*
  * Allows or disallows the FPGA from configuring. When disallowed,
  * initialization (erasing of configuration memory) takes place, but the FPGA
  * does not proceed to the configuration phase.
  */
 void permit_fpga_configuration(bool enable)
 {
-#if ((_BOARD_REVISION_MAJOR_ == 1) && (_BOARD_REVISION_MINOR_ > 2))
-	gpio_set_pin_level(FPGA_INITN, enable);
-	gpio_set_pin_direction(FPGA_INITN, GPIO_DIRECTION_OUT);
+#ifdef BOARD_HAS_FPGA_SYSCFG_PINS
```

Comment:
The outer (compile-time) condition could be eliminated if we remove `BOARD_HAS_FPGA_SYSCFG_PINS`.